src/Mvc/Mvc.Abstractions/src/ModelBinding/ModelMetadata.cs

```diff
@@ -95,6 +101,94 @@ public virtual ModelMetadata ContainerMetadata
         /// </summary>
         public abstract ModelPropertyCollection Properties { get; }
 
+        internal IReadOnlyList<ModelMetadata> BoundProperties
+        {
+            get
+            {
+                // An item may appear as both a constructor parameter and a property. For instance, in record types,
+                // each constructor parameter is also a settable property and will have the same name, possibly with a difference in case.
```

Comment:
Is this if there are more than one constructor? I thought the casing on the property and the parameter would match in a primary constructor

Reply:
The doc I linked to is older and https://github.com/dotnet/csharplang/blob/master/proposals/records.md#primary-constructor makes it clear(ish) users can't write their own primary constructor:
> It is an error to have a primary constructor and a constructor with the same signature already present in the class.

To @javiercn's point, src/Mvc/Mvc.Core/src/ModelBinding/Metadata/DefaultBindingMetadataProvider.cs won't bind unless there is just one constructor.